go/master/service.go

```diff
@@ -75,16 +82,26 @@ type Service struct {
 	chunksPerTask int
 	timeoutDur    time.Duration
 	failureMax    int
-	ready         chan struct{}
 	store         Store
 
+	ready    *sync.Cond
+	initDone bool
+
 	mu         sync.Mutex
-	initDone   bool
 	taskQueues taskQueues
+
+	passEndCond        *sync.Cond
+	clientCount        int
+	clientPassFinished int
 }
 
 func partition(chunks []Chunk, chunksPerTask int) []taskEntry {
-	id := 0
+	// generate uniq id across job using nanosecond + randint + counter
+	// FIXME(typhoonzero): this is a workaround, use uuid
+	randStart := rand.Int()
+	counter := 0
+	timestamp := time.Now().Nanosecond()
+	id := timestamp + randStart + counter
```

Comment:
With multiple datasets, task id may duplicate since `partition` will be called multiple times.